### ISSUE TITLE
Units of the .Moments and .nBound

### ISSUE THREAD

**Issue by 20190111** [shankar1729/jdftx#128]
Dear  Shankar,
I would like to ask you some questions about JDFT software.
1) I got the .n, .nbound, .Moments files after the fix-potential calculation, I know the unit of data in the .n file is the electron/(Bohr^3), but What's the unit for the date in the .nbound file and the .Moments ?  
2) What's the meaning of a nagative value for the total moments along z axis ? Does it mean that the direction of the dipole moment is from the surface to the bulk metallic (i.e., to the negative z axis)? 
3) The unit of the density of states in the manual is electrons/UnitCell/Hartree, I found that the data values for y axis are so big. I also calculate the dos for the same structure with VASP software, but I got much small values for density of states with the normal unit of states/eV. How can I change the unit of electrons/UnitCell/Hartree to the normal states/eV ?
4) If I want to get the d orbital DOS of the atom Fe in the strcture, I can set the parameter with OrthoOrbital Fe 1 d , but how can I get the dos of dz2 for the same Fe atom ?  
5) We do differential charge density calculations by AB.n - A.n -B.n without fix-potential condition, but how can I do the differential charge density calculations with fix-potential calculations(above PZC 0.5V for example) ?  When I set the same mu for AB, A and B, I found that the electrons in AB.out  is not equal to the number of electrons for A.out+B.out, but for a differential charge density calculation, we need the electrons in AB.n is equal to A.n+B.n. So,  Is my parameter setting wrong ? Please correct me and teach me how to do it ? 

Best 
Anna

**Comment by shankar1729** [comment]
Hi Anna,

1. All of them are in appropriate atomic units: e/bohr^3 for n and nbound; e-bohr for Moments.

2. The dipole moment is a vector, so the sign does indicate direction. (Just note the e-is-positive sign convention.)

3. You may be seeing the near-delta functions from the semi-core electrons. Focus the plot in on the relevant energy range first. Also recall our previous discussion in [Issue 98](https://github.com/shankar1729/jdftx/issues/98). Converting states/Hartree to states/eV just requires diving by 27.21.

4. Use "OrthoOrbital Fe 1 dz2". The dos command documentation lists the possible orbital / orbital set choices.

5. The differential charge density will not be net neutral for a fixed potential calculation: that is the point! Essentially charge is no longer conserved within the unit cell since it can flow in from the reservoir. So proceed as always, but expect a net charge.

Best,
Shankar

**Comment by 20190111** [comment]
Hi Shankar,
Thanks for your reply. But I still don't quite understand the answer to question 5. I know that " The differential charge density will not be net neutral for a fixed potential calculation", and we assume that the number of electrons in the neutral AB structure is total, I can obtain the total number of electrons(named as total1) in the AB1 .out after  a fixed potential calculation(the charged structure named as AB1, and the fix-potential is mu), then the net charge for AB1 is (total-total1).  I applied the same parameters of AB1 to A and B with the fixed potential calculation, but without ionic optimization, then I check the A.out and B.out, get the number of electrons a, b for the charged A and B.  But now I find that  a+b is not equal to total1. If the whole number of electrons in the AB1.n is equal to the nElectrons in AB1.out, then I could not do the differential charge density calculation with AB1.n-A.n -B.n(A.n abd B.n are the files obtained by fixed potential calculation with a single point energy calculation under mu), because the numer of electrons for A.n+B.n is not equal to AB1.n .  Please correct me if I am wrong and tell me what to do ?

Best Anna

**Comment by shankar1729** [comment]
Hi Anna,

The point of the fixed potential calculations are that their charge state is variable. The charge distribution on a metal surface with some adsorbate is not going to be net neutral at an arbitrary potential. 

Why does the number of electrons in the difference AB1.n - A.n - B.n need to be zero? This is just the practice in fixed charge DFT calculations where it makes sense to compare at fixed total electron number. If you want to capture the electron density rearrangement for a fixed potential case, you have to allow for a net change in the electron number too.

Best,
Shankar

**Comment by 20190111** [comment]
Hi Shankar，
Thank you for your patient answer. I get your meaning. But If I make a molecule(A) adsorbed on the metal slab(B), I got the AB.n, A.n and B.n, AB.out,A.out,B.out. For the structure of AB and B, we know that they have fermi level, and we can get different charge distribution under different fixed potential calculations, but for the molecule(A), there is no fermi level(i.e. mu), in order to get the differential charge density, we should carry out the same calculation with the same mu with AB and B, and the JDFTx software  does help us with fixed potential calculations for the molecule. How do we understand the physics of this?

Another question is that what's the relationshp between the numbers of electron in AB.n, AB.out and AB.nbound after a fixed potential calculation ?  The origin of this problem is that I found that the  integrated number of electrons in AB.n is not equal to the nElectrons in the AB.out, and the sum  integrated electrons for AB.n and AB.nbound is still not equal to the  nElectrons in the AB.

Best 
Anna

**Comment by shankar1729** [comment]
Hi Anna,

For the molecules, you should not perform a  fixed potential calculation. Keep it neutral, as in the isolated molecule case, there will not be a charge reservoir to equilibriate with. I'm not sure what exactly you are trying to show in the charge difference plot: you need to define the reference condition with respect to which you want to show the charge difference first. In fact, a much clearer and easier charge difference plot may be to compare the fixed potential molecule adsorbed on surface against the neutral version of the same: that will show how the extra charge redistributes. So first decide what you are trying to visualize here.

The integrated number of the electron density file should be equal to nElectrons in the output file. If this is the case, please post those output files and we'll try to figure out the discrepancy.

Best,
Shankar

**Comment by 20190111** [comment]
Hi Shankar,
Thanks for your reply and I will think about your suggestion a little bit more.
[Test-TiO2.zip](https://github.com/shankar1729/jdftx/files/5012011/Test-TiO2.zip)
Here are the files which I try to compare the number of electrons. The integrated number of  electrons for sol-cold.n is 140.33, while the nElect in the sol-cold.out is 144. I can not figure out the discrepancy.

Best,
Shankar

**Comment by shankar1729** [comment]
Hi Anna,

Try the difference AB.n - A.n - B.n where AB is your fixed-potential calculation of the full system, A is a fixed-potential calculation of the surface alone at the same potential and B is an isolated calculation of your molecule not at fixed potential. The integral of this difference will not be zero, but that is irrelevant for the reasons we discussed in this thread previously.

I checked your files and the integrals of the n files match the reported electrons exactly. There must be some error in your integration routine / processing of the electron density.

Best,
Shankar
